src/utils/productutilFuncs.js

```diff
@@ -0,0 +1,30 @@
+const getProductClass = (product) => {
+    if (product.isNewArrival) {
+        if (product.isNewArrival && !product.isInStock) {
+            return "card-w-badge-type1 card-w-badge-type3 card-w-badge b-rad1"
+        } else {
+            return "card-w-badge-type1 card-w-badge b-rad1"
+        }
+    } else {
+        if (product.isTrending) {
+            if (product.isTrending && !product.isInStock) {
+                return "card-w-badge-type2 card-w-badge-type3 card-w-badge b-rad1"
+            } else {
+                return "card-w-badge-type2 card-w-badge b-rad1"
+            }
+        } else {
+            if (product.isInStock) {
+                return "card-w-badge-type1 card-w-badge b-rad1"
+            } else {
+                return "card-w-badge-type3 card-w-badge b-rad1"
+            }
+        }
+    }
+}
```

Comment:
This is good but you can try to remove some if statement. like in line 2 you have a check for `isNewArival` and you have the same check in line 3 also.

Reply:
yes, thanks for pointing it out!! will refactor the code here!!